app/uk/gov/hmrc/agentsubscriptionfrontend/service/TaskListService.scala

```diff
@@ -35,24 +35,69 @@ class TaskListService @Inject()(agentAssuranceConnector: AgentAssuranceConnector
       maa <- agentAssuranceConnector.isManuallyAssuredAgent(subscriptionJourneyRecord.businessDetails.utr)
     } yield {
       if (isCleanCredsAgent(subscriptionJourneyRecord)) {
-        val amlsTask = AmlsTask(maa, subscriptionJourneyRecord.amlsData)
-        val checkAnswersTask = CheckAnswersTask(amlsTask)
-        List(amlsTask, checkAnswersTask)
+        val amlsTask: Task = AmlsTask(List(AmlsSubTask(maa, subscriptionJourneyRecord.amlsData)))
+        val contactEmailSubTask: SubTask = ContactDetailsEmailSubTask(
+          subscriptionJourneyRecord.contactDetailsEmailCheck,
+          amlsTask.isComplete
+        )
+        val contactTradingNameSubTask: SubTask = ContactTradingNameSubTask(
+          subscriptionJourneyRecord.contactDetailsTradingName,
+          contactEmailSubTask.isComplete
+        )
+        val contactTradingAddressSubTask: SubTask = ContactTradingAddressSubTask(
+          subscriptionJourneyRecord.contactDetailsTradingAddress,
+          contactTradingNameSubTask.isComplete
+        )
+        val contactDetailsTask: Task = ContactDetailsTask(
+          List(
+            contactEmailSubTask,
+            contactTradingNameSubTask,
+            contactTradingAddressSubTask
+          )
+        )
```

Comment:
its all 20 lines of duplicated code in if and else blocks ...
can we refactor this ?